### PR TITLE
Add property CustomSvg to icons and navigation

### LINKED ISSUE #1422: Allow plugins to specify a custom icon
**Describe the problem/challenge you have**

Clarity has a nice generic set of computing related icons, but the odds of Clarity having an icon appropriate for a plugin, which are often tied to a specific project, is unlikely.

**Describe the solution you'd like**

Plugins should be able to define a custom navigation icon. For example, a Knative plugin should use the Knative logo, and not a generic cloud icon.

**Anything else you would like to add:**

Data URIs are a good way to avoid plugins needing to reference external assets. To aid theming and visual consistency, a single SVG stroke could be allowed that is then styled by the client to match the current theme.

### PR DESCRIPTION
**Pull request by ftovaro**
**What this PR does / why we need it**:

This allows plugin authors to setup their own personal SVG for their plugins

Screen Shot 2021-08-23 at 11 02 48 AM

**Which issue(s) this PR fixes**
- Fixes #1422 

**Special notes for your reviewer**:

**Release note**:
```
release-note
```

### CODE REVIEW

File: cmd/octant-sample-plugin/main.go [hunk 1]
```diff
@@ -188,6 +188,18 @@ func handlePrint(request *service.PrintRequest) (plugin.PrintResponse, error) {
 // The plugin can use whatever paths it likes since these paths can be namespaced to the
 // the plugin.
 func handleNavigation(request *service.NavigationRequest) (navigation.Navigation, error) {
+	log.Printf("Printing something in navigation")
```
Comment (xtreme-vikram-yadav) [inline cmd/octant-sample-plugin/main.go:191]: these print statements seem omissible.

File: web/src/stories/icon.stories.mdx [hunk 2]
```diff
@@ -64,6 +92,12 @@ i := component.NewIcon("user", component.WithTooltip("hello", component.TooltipM
 i.Config.Color = "#2596be"
 `}}
 
+export const iconWithSvgDocs = {source: {code: `
+i := component.NewIcon("my-custom-icon")
+svg := <svg width="10px" height="10px"><g><ellipse fill="#000000" stroke="#000000" cx="5" cy="5.5" rx="3" ry="3"/></g></svg>
```
Comment (GuessWhoSamFoo) [inline web/src/stories/icon.stories.mdx:97]: What if the width and height are larger sizes? Does the icon scale?

File: web/src/stories/icon.stories.mdx [hunk 3]
```diff
@@ -64,6 +92,12 @@ i := component.NewIcon("user", component.WithTooltip("hello", component.TooltipM
 i.Config.Color = "#2596be"
 `}}
 
+export const iconWithSvgDocs = {source: {code: `
+i := component.NewIcon("my-custom-icon")
```
Comment (GuessWhoSamFoo) [inline web/src/stories/icon.stories.mdx:96]: There is potential for a name collision with clarity icons (https://clarity.design/foundation/icons/shapes/) that results in showing an empty svg. Please document this behavior or use the client to render an error in such a case.

File: web/src/stories/docs/plugins/GoPluginsIntro.story.mdx [hunk 4]
```diff
@@ -327,3 +327,45 @@ func handlePrint(request *service.PrintRequest) (plugin.PrintResponse, error) {
   ctx = metadata.AppendToOutgoingContext(request.Context(), "x-octant-foo", "bar")
   u, err := request.DashboardClient.Get(ctx, key)
 ```
+
+## Custom Icon
+
+### Custom Icon for plugins
+
+Is possible to add a custom SVG for plugins.
```
Comment (GuessWhoSamFoo) [inline web/src/stories/docs/plugins/GoPluginsIntro.story.mdx:335]: Nit: grammar

File: web/src/app/modules/shared/services/navigation/navigation.service.ts [hunk 5]
```diff
@@ -186,7 +192,11 @@ export class NavigationService {
             path: module.path,
             icon: module.icon,
             title: module.title,
+            customSvg: module.customSvg,
           };
+          if (module.customSvg) {
+            ClarityIcons.addIcons([module.icon, module.customSvg]);
```
Comment (GuessWhoSamFoo) [inline web/src/app/modules/shared/services/navigation/navigation.service.ts:198]: What about navigation icons for children?

**Comment by ftovaro** [comment]
Note:

Is possible to add oversized icons with svg, the only thing needed is to set the size of the icon.

```
i := component.NewIcon("my-icon")
svg := `<svg width="300" height="200"><polygon points="100,10 40,198 190,78 10,78 160,198"style="fill:lime;stroke:purple;stroke-width:5;fill-rule:evenodd;" /></svg>`
i.SetCustomSvg(svg)
i.SetSize("200")
```

image

File: web/src/stories/icon.stories.mdx (hunk 2 above)
Comment (ftovaro) [reply web/src/stories/icon.stories.mdx:97]: It's possible to add oversized icons with svg, the only thing needed is to set the size of the icon.

```
i := component.NewIcon("my-icon")
svg := `<svg width="300" height="200"><polygon points="100,10 40,198 190,78 10,78 160,198"style="fill:lime;stroke:purple;stroke-width:5;fill-rule:evenodd;" /></svg>`
i.SetCustomSvg(svg)
i.SetSize("200")
```

image

File: web/src/stories/icon.stories.mdx [hunk 6]
```diff
@@ -53,6 +53,32 @@ export const iconWithTooltipView=  {
   }
 };
 
+export const svg = `
```
Comment (GuessWhoSamFoo) [inline web/src/stories/icon.stories.mdx:56]: The validation might be too aggressive. This is valid input currently being filtered out as seen in storybook

File: web/src/stories/icon.stories.mdx [hunk 7]
```diff
@@ -64,6 +90,13 @@ i := component.NewIcon("user", component.WithTooltip("hello", component.TooltipM
 i.Config.Color = "#2596be"
 `}}
 
+export const iconWithSvgDocs = {source: {code: `
+i := component.NewIcon("my-custom-icon")
+<svg width="300" height="200"><polygon points="100,10 40,198 190,78 10,78 160,198"style="fill:lime;stroke:purple;stroke-width:5;fill-rule:evenodd;" /></svg>
```
Comment (GuessWhoSamFoo) [inline web/src/stories/icon.stories.mdx:95]: Fix this code snippet

**Review by GuessWhoSamFoo — APPROVED** [review]

**Pull request merged**